app/script/media/MediaStreamHandler.js

```diff
@@ -194,7 +194,25 @@ z.media.MediaStreamHandler = class MediaStreamHandler {
    * @returns {Promise} Resolves when the MediaStream has been replaced
    */
   replaceMediaStream(mediaStreamInfo) {
-    const {stream: mediaStream, type} = mediaStreamInfo;
+    const {stream: mediaStream} = mediaStreamInfo;
+
+    const replaceMediaStream = (localMediaStream, newMediaStreamInfo) => {
+      this._releaseMediaStream(this.localMediaStream());
+      this._setStreamState(updateMediaStreamInfo);
+      this.localMediaStream(updateMediaStreamInfo.stream);
+    };
+
+    const replaceMediaTracks = (localMediaStream, newMediaStream) => {
+      localMediaStream.getTracks().forEach(localTrack => {
+        updateMediaStreamInfo.stream.getTracks().forEach(newTrack => {
+          if (localTrack.kind === newTrack.kind) {
+            localTrack.stop();
+            localMediaStream.removeTrack(localTrack);
+            localMediaStream.addTrack(newTrack);
+          }
+        });
+      });
+    };
```

Comment:
We already have this code. Let's not duplicate it.

Reply:
Where is it?

Reply:
Yeah ok I see. I have to admit this `clone` part kind of scares me. I definitly don't want to clone the stream in that case and I wonder what could be the side effects of this...